src/app/App.jsx

```diff
@@ -92,73 +89,77 @@ class App extends React.Component {
     this.handleBidiChange = this.handleBidiChange.bind(this);
     this.handleThemeChange = this.handleThemeChange.bind(this);
     this.handleLocaleChange = this.handleLocaleChange.bind(this);
+
+    this.utilityConfig = this.setupUtilityConfig(props.utilityConfig);
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.defaultLocale !== undefined && nextProps.defaultLocale !== this.props.defaultLocale) {
+    if (App.propExistsAndChanged(nextProps.defaultLocale, this.props.defaultLocale)) {
       document.getElementsByTagName('html')[0].setAttribute('lang', nextProps.defaultLocale);
       this.setState({ locale: nextProps.defaultLocale });
     }
 
-    if (nextProps.defaultTheme !== undefined && nextProps.defaultTheme !== this.props.defaultTheme) {
+    if (App.propExistsAndChanged(nextProps.defaultTheme, this.props.defaultTheme)) {
       this.setState({ theme: nextProps.defaultTheme });
     }
 
-    if (nextProps.defaultDir !== undefined && nextProps.defaultDir !== this.props.defaultDir) {
+    if (App.propExistsAndChanged(nextProps.defaultDir, this.props.defaultDir)) {
       document.getElementsByTagName('html')[0].setAttribute('dir', nextProps.defaultDir);
       this.setState({ dir: nextProps.defaultDir });
     }
+
+    this.utilityConfig = this.setupUtilityConfig(nextProps.utilityConfig);
```

Comment:
Do we need to setup the utility config each time the app receive props?

Reply:
Yes, this applies the onChange and metadata instance functions to the static config provided externally.